src/modules/components/WatchTask.js

```diff
@@ -19,25 +19,22 @@ function renderInContainer(props, node) {
 export default function WatchTask(props) {
   const {
     children,
-    task: {
-      state,
-      output,
-    },
+    task,
   } = props
 
-  if (state === ERROR || (state === SUCCESS && output.errors)) {
+  if (isError(props.task)) {
     return renderInContainer(props,
       <Text>
         The loading failed
       </Text>,
     )
   }
 
-  if (state === PROGRESS) {
+  if (task.state === PROGRESS) {
     return renderInContainer(props, <CircularProgress />)
   }
 
-  if (state === SUCCESS) {
+  if (task.state === SUCCESS) {
     return children()
```

Comment:
This component is not in the React way, I think we should transform it but for now good for me.